{"kind":"symbol","hierarchy":{"paths":[["doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK","doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate"]]},"schemaVersion":{"major":0,"patch":0,"minor":3},"abstract":[{"type":"text","text":"Called when an error occurs during TrustlessSDK operations."}],"variants":[{"paths":["\/documentation\/trustlesssdk\/trustlesssdkdelegate\/didencountererror(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"didEncounterError","kind":"identifier"},{"kind":"text","text":"("},{"kind":"externalParam","text":"_"},{"text":" ","kind":"text"},{"text":"error","kind":"internalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:12TrustlessSDK0A5ErrorO","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessError","text":"TrustlessError","kind":"typeIdentifier"},{"kind":"text","text":")"}],"platforms":["iOS"],"languages":["swift"]}]},{"kind":"parameters","parameters":[{"name":"error","content":[{"inlineContent":[{"type":"text","text":"A "},{"code":"TrustlessError","type":"codeVoice"},{"text":" indicating the type of error that occurred. Implement this method","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"to handle errors specific to TrustlessSDK operations, such as displaying error messages to the user or"},{"text":" ","type":"text"},{"type":"text","text":"attempting to recover from the error. This method is optional; if not implemented, no action is taken by default."}],"type":"paragraph"}]}]},{"content":[{"text":"Discussion","anchor":"discussion","type":"heading","level":2},{"style":"note","type":"aside","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":""},{"type":"codeVoice","code":"TrustlessError"},{"text":" is an enumeration that should define various error states that TrustlessSDK","type":"text"},{"text":" ","type":"text"},{"text":"might encounter (e.g., network errors, authentication failures).","type":"text"}]}],"name":"Note"}],"kind":"content"}],"metadata":{"fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"didEncounterError","kind":"identifier"},{"kind":"text","text":"("},{"text":"TrustlessError","kind":"typeIdentifier","preciseIdentifier":"s:12TrustlessSDK0A5ErrorO"},{"kind":"text","text":")"}],"required":true,"roleHeading":"Instance Method","modules":[{"name":"TrustlessSDK"}],"title":"didEncounterError(_:)","role":"symbol","symbolKind":"method","externalID":"s:12TrustlessSDK0A11SDKDelegateP17didEncounterErroryyAA0aF0OF"},"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate\/didEncounterError(_:)"},"sections":[],"references":{"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK":{"title":"TrustlessSDK","url":"\/documentation\/trustlesssdk","abstract":[],"kind":"symbol","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK","role":"collection","type":"topic"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessSDKDelegate":{"kind":"symbol","abstract":[{"type":"codeVoice","code":"TrustlessSDKDelegate"},{"text":" protocol defines the delegate methods that ","type":"text"},{"type":"codeVoice","code":"TrustlessSDK"},{"type":"text","text":" uses to communicate"},{"type":"text","text":" "},{"type":"text","text":"with its delegate object. Implement these methods to handle session expiration and errors specifically"},{"text":" ","type":"text"},{"text":"related to the TrustlessSDK operations.","type":"text"}],"navigatorTitle":[{"kind":"identifier","text":"TrustlessSDKDelegate"}],"url":"\/documentation\/trustlesssdk\/trustlesssdkdelegate","type":"topic","fragments":[{"kind":"keyword","text":"protocol"},{"text":" ","kind":"text"},{"text":"TrustlessSDKDelegate","kind":"identifier"}],"title":"TrustlessSDKDelegate","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate","role":"symbol"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessSDKDelegate/didEncounterError(_:)":{"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate\/didEncounterError(_:)","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"didEncounterError"},{"kind":"text","text":"("},{"preciseIdentifier":"s:12TrustlessSDK0A5ErrorO","text":"TrustlessError","kind":"typeIdentifier"},{"text":")","kind":"text"}],"url":"\/documentation\/trustlesssdk\/trustlesssdkdelegate\/didencountererror(_:)","required":true,"type":"topic","title":"didEncounterError(_:)","kind":"symbol","abstract":[{"type":"text","text":"Called when an error occurs during TrustlessSDK operations."}],"role":"symbol"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessError":{"title":"TrustlessError","url":"\/documentation\/trustlesssdk\/trustlesserror","abstract":[{"type":"text","text":"Describes errors that can occur within the TrustlessSDK."}],"kind":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"TrustlessError","kind":"identifier"}],"navigatorTitle":[{"kind":"identifier","text":"TrustlessError"}],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessError","role":"symbol","type":"topic"}}}